### PR TITLE
Add new rule dir_perms_world_writable_system_owned_group.

### PR DESCRIPTION
**Pull request by ggbecker**
### Description:
- Add new rule `dir_perms_world_writable_system_owned_group`.

- Change old STIG reference ID from dir_perms_world_writable_system_owned
because this rule actually checks for UID and not the GID as it was
expected.

#### Rationale:

- There is a new requirement and I believe there was a confusion before in our rule that was supposed to check the group owner (gid) but was checking the user owner (uid). This pull request addresses this confusion as well.

- Group owner: https://www.stigviewer.com/stig/red_hat_enterprise_linux_7/2020-09-03/finding/V-204487
- User owner: https://www.stigviewer.com/stig/red_hat_enterprise_linux_7/2020-09-03/finding/V-228563

### CODE REVIEW

**Comment by redhatrises** [comment]
@openscap-ci test this please

**Comment by redhatrises** [comment]
CCE needs to be updated.

**Comment by ggbecker** [comment]
> CCE needs to be updated.

Done by force pushing

**Comment by ggbecker** [comment]
@openscap-ci test this please

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    <unix:object object_ref="all_local_directories_gid" />
+    <unix:state state_ref="state_gid_is_user_and_world_writable" />
+  </unix:file_test>
+  <unix:file_object comment="all local directories" id="all_local_directories_gid" version="1">
+    <unix:behaviors recurse="directories" recurse_direction="down" max_depth="-1" recurse_file_system="local" />
+    <unix:path operation="equals">/</unix:path>
+    <unix:filename xsi:nil="true" />
+    <filter action="include">state_gid_is_user_and_world_writable</filter>
+  </unix:file_object>
+  <unix:file_state comment="gid greater than or equal to {{{ auid }}} and world writable" id="state_gid_is_user_and_world_writable" version="1">
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml:18]: Nitpick: `auid` should probably be `uid_min`.

At the moment, by default, `auid` is set to `uid_min` when not explicitly defined.
If one would want to customize the starging range audited users (`auid`), this rule would change behavior too (I admit I haven't heard o such use case).

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml [hunk 2]
```diff
@@ -6,16 +6,16 @@
     </criteria>
   </definition>
   <unix:file_test check="all" comment="check for local directories that are world writable and have uid greater than or equal to {{{ auid }}}" id="test_dir_world_writable_uid_gt_value" version="1">
-    <unix:object object_ref="all_local_directories" />
-    <unix:state state_ref="state_gid_is_user_and_world_writable" />
+    <unix:object object_ref="all_local_directories_uid" />
+    <unix:state state_ref="state_uid_is_user_and_world_writable" />
```
Comment (yuumasato) [inline linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml:10]: Another good finding! :)

**Review by yuumasato — APPROVED** [review]
Looks good to me.
@ggbecker I'll leave it up to you consider my nitpick or not, :)

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml (hunk 1 above)
Comment (ggbecker) [reply linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml:18]: I'm not sure exactly what to decide.

Do you think that the same applies for? https://github.com/ComplianceAsCode/content/pull/6421/files/c11311736558613b13ae051a2908c31eee0b6a43#diff-a840b889a179d9be9756693119366b0e213700be3da9341baa6b5644905f1eccR18

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml (hunk 2 above)
Comment (ggbecker) [reply linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned/oval/shared.xml:10]: ;)

File: linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml (hunk 1 above)
Comment (yuumasato) [reply linux_os/guide/system/permissions/files/dir_perms_world_writable_system_owned_group/oval/shared.xml:18]: Yes.

**Pull request merged**